analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -210,5 +211,22 @@ private ProgramState ResetFieldConstraints(Func<IFieldSymbol, bool> predicate)
             }
             return state;
         }
+
+        [Conditional("DEBUG")]
+        public void CheckStateConsistency()
+        {
+            Debug.Assert(SymbolValue.Values.All(CheckConstraintAlsoHasNotNull<BoolConstraint>), "If a BoolConstraint is set for a symbol, NotNull should also be set.");
+            Debug.Assert(SymbolValue.Values.All(CheckConstraintAlsoHasNotNull<LockConstraint>), "If a LockConstraint is set for a symbol, NotNull should also be set.");
+            Debug.Assert(SymbolValue.Values.All(x => CheckOnlyConstraint(x, ObjectConstraint.Null)), "If Null is set for a symbol, no other constraint should be set.");
+            Debug.Assert(OperationValue.Values.All(CheckConstraintAlsoHasNotNull<BoolConstraint>), "If a BoolConstraint is set for a operation, NotNull should also be set.");
+            Debug.Assert(OperationValue.Values.All(CheckConstraintAlsoHasNotNull<LockConstraint>), "If a LockConstraint is set for a operation, NotNull should also be set.");
+            Debug.Assert(OperationValue.Values.All(x => CheckOnlyConstraint(x, ObjectConstraint.Null)), "If Null is set for a operation, no other constraint should be set.");
```

Comment:
This set of fat assertions looks like a 2x3 duplication. Can you extract it, or am I missing something?